tests/consumers/duckdb_consumer.py

```diff
@@ -0,0 +1,64 @@
+import os
+import string
+from collections.abc import Iterable
+
+import pyarrow as pa
+
+
+class DuckDBConsumer:
+    """
+    Implementation of the DuckDB substrait consumer Class for testing
+    """
+
+    @staticmethod
+    def run_substrait_query(db_connection, sql_query: str) -> pa.Table:
+        """
+        Convert the SQL query into a substrait query plan and run the plan against DuckDB.
+
+        Parameters:
+            db_connection:
+                DuckDB connection.
+            sql_query:
+                A SQL query string.
+
+        Returns:
+            A pyarrow table resulting from running the substrait query plan.
+        """
+        duckdb_substrait_plan = db_connection.get_substrait(sql_query)
+        proto_bytes = duckdb_substrait_plan.fetchone()[0]
+        return db_connection.from_substrait(proto_bytes).arrow()
+
+    @staticmethod
+    def load_tables_from_parquet(
+        db_connection,
+        created_tables: set,
+        file_names: Iterable[str],
+        parquet_file_paths: Iterable[str],
+    ) -> list:
+        """
+        Load all the parquet files into separate tables in DuckDB.
+
+        Parameters:
+            db_connection:
+                DuckDB connection.
+            created_tables:
+                The set of tables that have already been created.
+            file_names:
+                Name of parquet files.
+            parquet_file_paths:
+                Full path of parquet files.
+
+        Returns:
+            A list of the table names.
+        """
+        table_names = []
+        for file_name, file_path in zip(file_names, parquet_file_paths):
+            table_name = os.path.splitext(file_name)[0]
```

Comment:
small nit -- I think we settled on `pathlib` in the initial commit -- do we want to keep that consistent?  I believe you can use the `stem` or `suffix` attributes of a `pathlib.Path` in place of `splitext()`

Reply:
Yeah, we did.  Just updated!